examples/chat-typing-indicator/page.md

```diff
@@ -0,0 +1,93 @@
+
+Typing indicators can be implemented using Ably Chat. The Chat SDK contains a set of purpose-built APIs that abstract away the complexities involved in architecting chat features. It  is built on top of Ably's core platform, and so it provides the same performance and scaling guarantees.
+
+## Resources
+
+// React brief (Only visible if viewing the React example)
+
+Use the following components to add typing indicators into a chat application:
+
+[ChatClientProvider](https://ably.com/docs/chat/setup#instantiate): initializes and manages a shared chat client instance, passing it down through React context to enable real-time chat functionality across the application.
+[ChatRoomProvider](https://ably.com/docs/chat/rooms#create): manages the state and functionality of a specific chat room, providing access to messages, participants, and real-time interactions within that room via React context.
+[useRoom()](https://ably.com/docs/chat/rooms#create) hook: a hook to manage the state and interaction for a chat “room”, allowing users to join, send messages, listen for messages, and use the other chat functions such as typing indicators.
+[useTyping()](https://ably.com/docs/chat/rooms/typing#subscribe) hook: a hook to manage and track the typing status of users within a chat room.
+
+For further information on the typing indicator feature please refer to the [typing indicator documentation](https://ably.com/docs/chat/rooms/typing).
+
+// End React brief
+
+// Javascript brief (Only visible if viewing the Javascript example)
+
+Use the following methods to add typing indicators into a chat application:
+
+[`rooms.get()`](https://ably.com/docs/chat/rooms#create) - creates a new or retrieves an existing `room`.
+[`rooms.typing.subscribe()`](https://ably.com/docs/chat/rooms/typing) - subscribes to typing events by registering a listener. Typing events are emitted when a user starts typing, or when they stop typing.
+[`room.typing.get()`](https://ably.com/docs/chat/rooms/typing#retrieve) - Retrieve list of users currently typing by their clientId.
+[`room.typing.start()`](https://ably.com/docs/chat/rooms/typing#set) - Emit a typing event that the user is currently typing, initialising the timeout which will call `room.typing.stop()` if no further events are emitted by the user.
+
+For further information on the typing indicator feature please refer to the [typing indicator documentation](https://ably.com/docs/chat/rooms/typing).
+
+// End Javascript brief
+
+## View on Github
+
+// React
+
+The full source code and instructions for getting started with can be found in Github.
+
+Once the repository has been cloned, follow the instructions below to run the application.
+
+Rename `.env.example` to `.env.local` and fill in your Ably API KEY in the `NEXT_PUBLIC_ABLY_KEY` environment variable.
```

Comment:
```suggestion
Rename `.env.example` to `.env.local` and enter your Ably API key in the `NEXT_PUBLIC_ABLY_KEY` environment variable.
```

Should we number these steps since they're instructions?

Reply:
I did wonder this before, but it's a really short piece of instruction. Especially when the content filtered out based on whether it's JS or React.  What would it be?

1 - Clone repo
2 - install deps
3 - run dev server
4 - open two tabs and interact.

What do you think?

Reply:
I'm not overly opinionated, but thought it might breakup the text a little. I think we'd get to at least 6 steps too, no? 

1. Clone repo
2. Change directory
3. Update API key
4. Install deps
5. Run server
6. Open tabs and interact?